deployment/utils/reverse-proxy.ts

```diff
@@ -14,6 +14,45 @@ export class Proxy {
     private staticIp?: { address?: string; aksReservedIpResourceGroup?: string },
   ) {}
 
+  registerInternalProxy(
+    dns: { record: string },
+    routes: {
+      path: string;
+      service: k8s.core.v1.Service;
+      upstreamUrl: string;
+      host: string; // use env helpers of course
+    }[],
+  ) {
+    new k8s.apiextensions.CustomResource(`internal-proxy-${dns.record}`, {
+      apiVersion: 'projectcontour.io/v1',
+      kind: 'HTTPProxy',
+      metadata: {
+        name: `internal-proxy-metadata-${dns.record}`,
+      },
+      spec: {
+        virtualhost: {
+          fqdn: 'lab-worker.app.graphql-hive.com',
+        },
+        routes: routes.map(route => ({
+          conditions: [{ prefix: route.path }],
+          services: [
+            {
+              name: route.service,
+              port: 1234,
```

Comment:
this should be calculated based on the `Service` you pass here. 

Check how it's done in the rest of the file: 

```
services: [
              {
                name: route.service.metadata.name,
                port: route.service.spec.ports[0].port,
              },
            ],
```